edn_format/edn_dump.py

```diff
@@ -60,17 +60,51 @@ def seq(obj, **kwargs):
     return ' '.join([udump(i, **kwargs) for i in obj])
 
 
+def _indent_objs(objs, open_symbol, close_symbol, indent, indent_step, **kwargs):
+    indent_prev = indent_step
+    indent_step = indent_prev + indent
+
+    result = '{}\n'.format(open_symbol)
+
+    for obj in objs:
+        if isinstance(obj, (dict, ImmutableDict)):
+            pairs = obj.items()
+            result += '{}{}\n'.format(
+                ' ' * indent_step,
+                _indent_objs(pairs, '{', '}', indent=indent, indent_step=indent_step, **kwargs)
+            )
+        elif isinstance(obj, (tuple, list, ImmutableList, set, frozenset)):
+            result += '{}{}\n'.format(
+                ' ' * indent_step,
+                seq(obj, indent=indent, indent_step=indent_step, **kwargs)
+            )
+        else:
+            result += '{}{}\n'.format(
+                ' ' * indent_step,
+                udump(obj, indent=indent, indent_step=indent_step, **kwargs)
+            )
+
+    result += '{}'.format(' ' * indent_prev)
+    result += '{}'.format(close_symbol)
```

Comment:
Perf note: `+=` creates a new string every time. It would be better to use `StringIO`:
```python
# Python 2
try:
    from cStringIO import StringIO  # try the C module first, it’s more efficient
except ImportError:
    from StringIO import StringIO
```
```python3
# Python 3:
from io import StringIO
```
then:
```python3
result = StringIO()
result.write(…) # append a string
result.write(…) # another one
# … 
return result.getvalue()
```